### ISSUE TITLE
Opportunity / Shared Contact

### ISSUE THREAD

**Issue by paulator** [frappe/erpnext#21949]
## Description of the issue
I have a restricted user in CRM. It can only view its own Leads, Opportunities and Contacts. In addition there are some Leads, Opportunities and Contacts that are shared by a Sales Manager.

When the restricted user is in the shared Lead, and creates a new Opportunity, then the Contact-Person search field shows wrong Contact Person.

## Context information (for bug reports)

**Output of `bench version`**
```
ERPNext: v12.9.3
Frappe: v12.6.1
```

## Steps to reproduce the issue

1. Create a new Lead with a Sales Manager user (an see all Leads)
2. Share it (read write) with everybody
3. Open that Lead with a restricted user that can only see its own Leads (should see it because of the sharing)
4. Click on "Create Opportunity"
5. Click in Contact Info section in Contact Person field, remove content and search for a Contact

### Observed result
1. First I can see one Contact from a different Lead (This contact is also shared with the restricted user)

2. I can search for all shared contacts in the field
![image](https://user-images.githubusercontent.com/7262648/82947869-045d5d00-9fa1-11ea-8f4e-84926c93c9b9.png)

3. The details of the Contact repeats the details
![image](https://user-images.githubusercontent.com/7262648/82947729-c5c7a280-9fa0-11ea-8895-0d53889a1321.png)

4. When I click on "Advanced Search" I get a infinite list repeating the the same contact again and again
![image](https://user-images.githubusercontent.com/7262648/82947589-99138b00-9fa0-11ea-8f3f-711b711f7ced.png)

### Expected result
1. I want to see only contacts for that Source-Lead (resp. Source Customer)
2. I want to have search results only for the matching Source Lead / Customer
3. I don't want to see the repeating details of the search result in the drop down field
4. I want to see the search result only once instead of repeating it
### Stacktrace / full error message

```
There is no stacktrace
```

## Additional information
Same behavior is in "Customer/Lead Address" field

ERPNext Docker Environment

**Comment by deepeshgarg007** [comment]
@paulator Closing this issue, if it still persists in the latest version-13 release or higher then please reopen a new issue
